docs/enhancements/0001-worker-pods.md

```diff
@@ -0,0 +1,229 @@
+                                                        │                       │           │ NodeModulesState │
+                                                        │ ┌──────────────────┐  │           │                  │
+              ┌───────────┐                             │ │                  │  │           └─────────▲────────┘
+              │           │        creates & owns;      │ │ NodeModulesState │  │                     │
+              │ ConfigMap ◄─────────────────────────────┼─┤ controller       ├──┼─────────────────────┘
+              │           │   writes worker config &    │ │                  │  │   watches &
+              └─────▲─────┘   reads worker output from  │ └──┬───────────────┘  │   updates status
+                    │                                   │    │                  │
+            mounts; │                                   │    │     KMM operator │
+reads config from & │       ┌────────────┐              └────┼──────────────────┘
+   writes output to │       │            │                   │
+                    └───────┤ KMM worker ◄───────────────────┘
+                            │    Pod     │ creates & owns
+                            │            │
+                            └──────┬─────┘
+                                   │
+                                   │
+                                   │               ┌──────────────────────┐
+                                   │               │                      │
+                                   └───────────────► registry.example.org │
+                                      pulls from   │                      │
+                                                   └──────────────────────┘
+```
+
+#### The operator
+
+The operator runs two main controllers: the `Module` controller and the `NodeModulesState` controller.
+
+The `Module` controller watches both `Module` and `Node` resources to determine which kmods should be loaded on which
+`Node`.
+It creates one `NodeModulesState` resource per `Node` in the cluster that is targeted to load at least one kmod.  
+`NodeModulesState` resources represent the desired state of all kmods that a certain node should have loaded; they
+contain all the required data to pull kmod images and load the modules properly.  
+The `Module` controller watches updates to the `status` of `NodeModulesState` objects and aggregates that data into the
+`Module` `status`.
+
+The `NodeModulesState` controller watches `NodeModulesState` resources and determines when a worker Pod needs to be
+started on a node by looking at the difference between the desired state (spec) and the current one (status).  
+It prepares a worker configuration file in a `ConfigMap` based on the actions that need to be run on the node and
+creates a Pod mounting that `ConfigMap`.  
+When the Pod has completed, the controller reads the worker's output from the same `ConfigMap` and updates the
```

Comment:
If we are going in the direction of creating Pod per kmod/node, do we still want ConfigMap? Would not it be simpler to just pass all the data as arguments to in the Pod definition itself?

Reply:
We are going to create a generic container image that will be the Image of worker pods. In case we are going with the option of Pod per Node per Module, and we are not going to output the result to ConfigMap, what's the point of creating ConfigMap per Pod just for input parameters sake, when we can just as easily code the parameters into the Pod itself?

Reply:
I didn't know we are ditching the output file. In this case I also think we can remove the config map, since we have a different config for each pod (unlike app that load the static config once in its lifetime) passing arguments/ENV variables to the container seems more suitable to me.